### ISSUE TITLE
Color and accessibility

### ISSUE THREAD

**Issue by jrpool** [carbon-design-system/carbon-website#2337]
## Summary

Some important distinctions are being made solely by changes in background color.

## Relevant information

The primary, secondary, and danger buttons differ from one another only in background color. The distinctions among them are significant, and the distinction between the danger button and the other two buttons is particularly important.

The claim for WCAG conformity appears to be based on per-button testing, but not on whole-document testing that could evaluate whether the document uses only color differences to make distinctions. WCAG 2.1 Success Criterion 2.4.1 states that “Color is not used as the only visual means of conveying information, indicating an action, prompting a response, or distinguishing a visual element.” The Accessibility Guidelines Working Group cites the example of making text red to indicate that the text describes an error. It seems reasonable to infer that making a button red to indicate that the button poses a danger is in the same category.

Given the apparent conflict between the color-only button distinctions and WCAG 2.1, it would be helpful for the documentation on this component, and any other similarly affected components, to include a discussion of the arguments for considering this design accessible.

**Comment by aagonzales** [comment]
Hi @jrpool, we believe color isn't the only visual difference, the label itself also acts as an identifier. If a user cannot tell the difference between buttons only from its color then the label should give them the required information needed to understand the severity. Additionally most danger actions are followed up with a [confirmation modal](https://carbondesignsystem.com/community/patterns/remove-pattern#high-impact-deletion) asking for certainty on the destructive action which gives the user an additional chance to understand the severity of the action. 

With Buttons there is also an optional additional visual indicator add an icon with the button. The icon can help identify the type of action being taken. 

I will bring this to the attention of our own accessibility experts but I don't believe further distinction is needed. @mbgower

**Comment by jrpool** [comment]
Thanks, @aagonzales, for your explanation. The rationale there is, as I understand it, that the implementer has the responsibility of communicating the danger in words, not relying on the button’s style to do that. The temptation to rely could be great, so reminding implementers not to do so might be helpful.

**Comment by mbgower** [comment]
I missed the prompt two years ago to comment on this, but noticed it being closed now, so thought I'd take the opportunity to support @aagonzales' basic response.

There are several distinctions to point out with the use of colour in Carbon buttons:
1. Colour is not differentiating operation. Operationally, all the buttons are the same. That is, they are all navigated and activated in the same way, regardless of colour. 
1. Colour is not solely differentiating function. The button's function is dependent on the button's label, not its color. That is, the text on the button (or the ALT of an icon-only button) describes the action it performs.
1. Although it is true that Carbon guides users to use red buttons for destructive actions, the use of colour remains complementary to the label. For instance, if a user coloured a Delete button blue, _Carbon_ may think it's wrong from their usage guidance, but there is no intrinsic accessibility failure. (Obviously if a user calls a button "Save" and the button deletes instead, that is a very real problem, but it is unrelated to color of the button.)
1. Carbon also reinforces the destructive nature of most delete functions with a confirmation dialog that appears after their activation. Note that this [confirmation is only required for WCAG](https://www.w3.org/WAI/WCAG21/Understanding/error-prevention-legal-financial-data.html) where the transaction involves more consequential deletions.
